### PR TITLE
Show error if Lemma is missing

### PR DESCRIPTION
**Pull request by micgro42**
This is the second PR split out of #192 and building on top of #198. It includes an entire vertical slice of the functionality

### CODE REVIEW

File: src/store/RootState.ts [hunk 1]
```diff
@@ -5,11 +5,18 @@ export interface SubmitError {
...
+
 export default interface RootState {
 	lemma: string;
 	language: SearchedItemOption | null;
 	languageCodeFromLanguageItem: string | undefined | null | false;
 	lexicalCategory: SearchedItemOption | null;
 	spellingVariant: string;
 	globalErrors: SubmitError[];
+	perFieldErrors: {
+		lemmaErrors: PerFieldError[];
```
Comment (lucaswerkmeister) [inline src/store/RootState.ts:20]: Do these errors get cleared anywhere?

Comment (micgro42) [reply src/store/RootState.ts:20]: Huh, I was so enamored with _adding_ those errors, that I fully forgot about clearing them! Will add the functionality right away!

Comment (micgro42) [reply src/store/RootState.ts:20]: Added it. Not sure how to best test it though :thinking:

**Review by lucaswerkmeister — COMMENTED** [review]
I just realized that this is different from the plan of action in T305854#7937644: we originally planned to check the lemma length on `input` events on the input, not just when the submit button is clicked…

**Comment by micgro42** [comment]
> I just realized that this is different from the plan of action in [T305854#7937644](https://phabricator.wikimedia.org/T305854#7937644)#7937644: we originally planned to check the lemma length on input events on the `input`, not just when the submit button is clicked…

But that comment explicitly discusses checking for too long a lemma and an empty lemma in two different sections, or am I misreading it?

**Comment by lucaswerkmeister** [comment]
> > I just realized that this is different from the plan of action in [T305854#7937644](https://phabricator.wikimedia.org/T305854#7937644)#7937644: we originally planned to check the lemma length on input events on the `input`, not just when the submit button is clicked…
> 
> But that comment explicitly discusses checking for too long a lemma and an empty lemma in two different sections, or am I misreading it?

You’re right, I got confused and thought this PR was already checking the length.

Is the length check not implemented at all yet?

**Comment by micgro42** [comment]
> Is the length check not implemented at all yet?

I think that part was not yet started to avoid conflicts with this work here.

File: src/store/mutations.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export const SET_LEXICAL_CATEGORY = 'setLexicalCategory';
 export const SET_SPELLING_VARIANT = 'setSpellingVariant';
 export const SET_LANGUAGE_CODE_FROM_LANGUAGE_ITEM = 'setLanguageCodeFromLanguageItem';
 export const ADD_ERRORS = 'addErrors';
+export const ADD_PER_FIELD_ERROR = 'addPerFieldError';
+export const CLEAR_PER_FIELD_ERROR = 'clearPerFieldError';
```
[the remark this answers is not in the document]
Comment (micgro42) [reply src/store/mutations.ts]: Done.

File: src/store/RootState.ts (hunk 1 above)
Comment (lucaswerkmeister) [reply src/store/RootState.ts:20]: I guess that feels like an integration test? But we could also do that at the end of the chain, for all inputs at once (errors are added on submit and cleared one by one as input is added).

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**